src/pages/places.tsx

```diff
@@ -159,81 +166,103 @@ export default function IndexPage() {
     [params, track, offset]
   )
 
-  const handleChangePois = useCallback(
-    (e: React.SyntheticEvent<any>, props: { value?: any }) => {
-      e.preventDefault()
-      e.stopPropagation()
-      const newParams = {
-        ...params,
-        only_featured: false,
-        only_pois: !!props.value,
-      }
-
-      if (newParams.order_by && newParams.only_pois) {
-        newParams.order_by = PlaceListOrderBy.LIKE_SCORE_BEST
-      }
-
+  const handleChangeOrder = useCallback(
+    (_: React.SyntheticEvent<any>, props: { value?: any }) => {
+      const value =
+        oneOf(props.value, getPlaceListQuerySchema.properties.order_by.enum) ??
+        PlaceListOrderBy.LIKE_SCORE_BEST
+      const newParams = { ...params, order_by: value, page: 1 }
       setAllPlaces([])
       track(SegmentPlace.FilterChange, {
         filters: newParams,
-        place: SegmentPlace.PlacesChangePois,
+        place: SegmentPlace.PlacesChangeOrder,
       })
       navigate(locations.places(newParams))
     },
     [params, track]
   )
 
-  const handleChangeFeatured = useCallback(
-    (e: React.SyntheticEvent<any>, props: { value?: any }) => {
-      e.preventDefault()
-      e.stopPropagation()
-      const newParams = {
-        ...params,
-        only_pois: false,
-        only_featured: !!props.value,
+  const searchParams = useMemo(
+    () => new URLSearchParams(location.search),
+    [location.search]
+  )
+
+  const handleSearchChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const newParams = new URLSearchParams(searchParams)
+      if (e.target.value) {
+        newParams.set("search", e.target.value)
+      } else {
+        newParams.delete("search")
       }
 
-      if (newParams.order_by && newParams.only_featured) {
-        newParams.order_by = PlaceListOrderBy.LIKE_SCORE_BEST
+      let target = location.pathname
+      const search = newParams.toString()
+      // location
+      // navigate to /search+=?search=${search}
+      if (search) {
+        target += "?" + search
       }
 
-      setAllPlaces([])
-      track(SegmentPlace.FilterChange, {
-        filters: newParams,
-        place: SegmentPlace.PlacesChangeFeatured,
-      })
-      navigate(locations.places(newParams))
+      navigate(target)
     },
-    [params, track]
+    [location.pathname, params]
   )
 
-  const handleChangeOrder = useCallback(
-    (_: React.SyntheticEvent<any>, props: { value?: any }) => {
-      const value =
-        oneOf(props.value, getPlaceListQuerySchema.properties.order_by.enum) ??
-        PlaceListOrderBy.LIKE_SCORE_BEST
-      const newParams = { ...params, order_by: value, page: 1 }
-      setAllPlaces([])
-      track(SegmentPlace.FilterChange, {
-        filters: newParams,
-        place: SegmentPlace.PlacesChangeOrder,
-      })
-      navigate(locations.places(newParams))
-    },
-    [params, track]
-  )
+  function handleCategorySelection(action: "add", categoryId: string): void
```

Comment:
All this behavior could be encapsulated inside a component `CampaignFilters` where you only listen the changed with eventHandler

Reply:
CampaignFilters 🤔